boa/src/builtins/math/mod.rs

```diff
@@ -717,16 +729,21 @@ impl Math {
     /// [spec]: https://tc39.es/ecma262/#sec-math.round
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round
     pub(crate) fn round(_: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
-        Ok(args
+        let num = args
             .get_or_undefined(0)
             //1. Let n be ? ToNumber(x).
-            .to_number(context)?
-            //2. If n is NaN, +∞𝔽, -∞𝔽, or an integral Number, return n.
-            //3. If n < 0.5𝔽 and n > +0𝔽, return +0𝔽.
-            //4. If n < +0𝔽 and n ≥ -0.5𝔽, return -0𝔽.
-            //5. Return the integral Number closest to n, preferring the Number closer to +∞ in the case of a tie.
-            .round()
-            .into())
+            .to_number(context)?;
+
+        //2. If n is NaN, +∞𝔽, -∞𝔽, or an integral Number, return n.
+        //3. If n < 0.5𝔽 and n > +0𝔽, return +0𝔽.
+        //4. If n < +0𝔽 and n ≥ -0.5𝔽, return -0𝔽.
+        //5. Return the integral Number closest to n, preferring the Number closer to +∞ in the case of a tie.
+
+        if (num.fract() + 0.5).abs() < f64::EPSILON {
```

Comment:
Is it absolutely necessary to compare with `f64::EPSILON`?
If the number has exactly 0.5 as a trailing fractional, then calculating `fract` should be lossless (because 0.5 is representable in f64), meaning we should be able to compare `num.fract() == 0.5` without a problem.
Am I overlooking something?

Reply:
Oh that's what you meant yeah I'm gonna fix this and squash the commits so that it's not too messy.